lib/cog/bootstrap.ex

```diff
@@ -54,6 +67,65 @@ defmodule Cog.Bootstrap do
     end)
   end
 
+  # Bootstraps using environment variables if they are present.
+  @doc """
+  Create an administrative user using values from the environment, if present.
+  If any of the following variables are not set, the bootstrap will not be
+  performed. Required variables:
+
+  COG_BOOTSTRAP_USERNAME
+  COG_BOOTSTRAP_PASSWORD
+  COG_BOOTSTRAP_EMAIL_ADDRESS
+  COG_BOOTSTRAP_FIRST_NAME
+  COG_BOOTSTRAP_LAST_NAME
+
+  Additionally, if the user bootstrap completes successfully and the RELAY_ID
+  and RELAY_COG_TOKEN variables are set, a relay will be created using the
+  variables and assigned to a default relay group, which will also be created.
+  If the relevant variables are not present, this step will be skipped.
+
+  If any creation attempt fails, the entire bootstrap process will be rolled
+  back.
+  """
+  def maybe_bootstrap do
+    unless is_bootstrapped? do
+      Repo.transaction fn ->
+        if bootstrap_from_env, do: relay_from_env(System.get_env)
+      end
+
+      true
+    end
+
+    false
+  end
+
+  def bootstrap_from_env do
+    user_keys = ~w(username password email_address first_name last_name)
+    user_map = for param <- user_keys do
+      key = "COG_BOOTSTRAP_" <> String.upcase(param)
+      { param, System.get_env(key) }
+    end |> Enum.into(%{})
+
+    unless Enum.any?(user_map, fn({_k, v}) -> v == nil end) do
```

Comment:
This might be a tiny bit more readable `user_map |> Map.values |> Enum.any?(&is_nil/1)`